models/app.model.js

```diff
@@ -4,4 +4,18 @@ exports.readTopics = () => {
     return db.query(`SELECT * FROM topics;`).then(({ rows }) => {
       return rows;
     })
-  }
\ No newline at end of file
+  }
+
+  exports.fetchArticleById = (article_id) =>{
+    if(isNaN(parseInt(article_id))){
+      return Promise.reject({ status: 400, msg: "Bad Request" });
+    }
```

Comment:
You don't need to do a manual check on the data type of article_id - if you try to query the database with a string for article id psql will throw an error itself. Then you can catch this in the catch block - send it to your error handler and check the code it gives you back.